saml-authentication-server/src/main/java/jetbrains/buildServer/auth/saml/plugin/SamlAuthenticationScheme.java

```diff
@@ -139,6 +144,22 @@ public HttpAuthenticationResult processAuthenticationRequest(@NotNull HttpServle
 
                             if (user == null) {
                                 LOG.warn(String.format("New user %s was not created due to unknown reason", username));
+                            } else {
+                                String groups = getAttribute(auth, settings.getGroupsAttributeMapping());
+                                LOG.debug(String.format("SAML Groups = '%s'", groups));
+
+                                // Look for matching userGroup
+                                String[] assignedGroups = groups.split(",");
```

Comment:
Error here - getAttribute joins collections with ", " (comma with space) not ",". Anyway, I think it's better to add some additional processing to the list - trim and filter empty records. Something like
```
Arrays.stream(groups.split(", "))
                                        .map(String::trim)
                                        .filter(StringUtil::isNotEmpty)
                                        .forEach(...);
```

Reply:
Addressed in [`732591b` (#52)](https://github.com/morincer/teamcity-plugin-saml/pull/52/commits/732591b0503ddf1447ad56eb81d5497aad25b4b0)